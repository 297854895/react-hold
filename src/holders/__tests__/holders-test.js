import React from 'react'
import { mount } from 'enzyme'
import { Circle, Fill, Square, Table, Text } from '../index'

describe('Holders', () => {
  it('render Circle correctly', () => {
    let wrapper = mount(<Circle>foo-0</Circle>)
    expect(wrapper).toMatchSnapshot()


    const props = {
      color: 'red',
      width: 50,
      height: 60,
    }
    wrapper = mount(<Circle {...props}>foo-0-1</Circle>)
    expect(wrapper).toMatchSnapshot()
  })

  it('render Fill correctly', () => {
    let wrapper = mount(<Fill>foo-1</Fill>)
    expect(wrapper).toMatchSnapshot()

    const props = {
      color: 'red',
      width: '60%',
      height: '30px',
    }
    wrapper = mount(<Fill {...props}>foo-1-1</Fill>)
    expect(wrapper).toMatchSnapshot()
  })

  it('render Square correctly', () => {
    let wrapper = mount(<Square>foo-2</Square>)
    expect(wrapper).toMatchSnapshot()

    const props = {
      color: 'red',
      width: 50,
      height: 60,
    }
    wrapper = mount(<Square {...props}>foo-2-1</Square>)
    expect(wrapper).toMatchSnapshot()
  })

  it('render Table correctly', () => {
    let wrapper = mount(<Table>foo-3</Table>)
    expect(wrapper).toMatchSnapshot()

    const props = {
      color: 'red',
      width: 360,
      height: 60,
      gap: 4,
    }
    wrapper = mount(<Table {...props}>foo-3-1</Table>)
    expect(wrapper).toMatchSnapshot()
  })

  it('render Text correctly', () => {
    let wrapper = mount(<Text>foo-4</Text>)
    expect(wrapper).toMatchSnapshot()

    const props = {
      color: 'red',
      length: 300,
    }
    wrapper = mount(<Text {...props}>foo-4-1</Text>)
    expect(wrapper).toMatchSnapshot()
  })
})
