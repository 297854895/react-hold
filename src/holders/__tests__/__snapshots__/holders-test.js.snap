// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Holders render Circle correctly 1`] = `
<Circle
  align="center"
  diameter={null}
  fillerStyle={null}
  height={null}
  width={null}
>
  <Square
    align="center"
    color="transparent"
    fillerStyle={null}
    height={null}
    side={null}
    width={null}
  >
    <div
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        style={
          Object {
            "background": "transparent",
            "display": "inline-block",
            "height": null,
            "lineHeight": null,
            "textAlign": "center",
            "width": null,
          }
        }
      >
        <div
          style={
            Object {
              "background": undefined,
              "borderRadius": "50%",
              "height": "100%",
              "width": "100%",
            }
          }
        >
          foo-0
        </div>
      </div>
    </div>
  </Square>
</Circle>
`;

exports[`Holders render Circle correctly 2`] = `
<Circle
  align="center"
  color="red"
  diameter={null}
  fillerStyle={
    Object {
      "color": "blue",
      "width": 40,
    }
  }
  height={60}
  width={50}
>
  <Square
    align="center"
    color="transparent"
    fillerStyle={null}
    height={60}
    side={null}
    width={50}
  >
    <div
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        style={
          Object {
            "background": "transparent",
            "display": "inline-block",
            "height": 50,
            "lineHeight": "50px",
            "textAlign": "center",
            "width": 50,
          }
        }
      >
        <div
          style={
            Object {
              "background": "red",
              "borderRadius": "50%",
              "color": "blue",
              "height": "100%",
              "width": "100%",
            }
          }
        >
          foo-0-1
        </div>
      </div>
    </div>
  </Square>
</Circle>
`;

exports[`Holders render Fill correctly 1`] = `
<Fill
  align="center"
  fillerStyle={null}
  height={null}
  width={null}
>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <div
      style={
        Object {
          "background": undefined,
          "display": "inline-block",
          "height": null,
          "lineHeight": null,
          "textAlign": "center",
          "width": null,
        }
      }
    >
      foo-1
    </div>
  </div>
</Fill>
`;

exports[`Holders render Fill correctly 2`] = `
<Fill
  align="center"
  color="red"
  fillerStyle={null}
  height="30px"
  width="60%"
>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <div
      style={
        Object {
          "background": "red",
          "display": "inline-block",
          "height": "30px",
          "lineHeight": "30px",
          "textAlign": "center",
          "width": "60%",
        }
      }
    >
      foo-1-1
    </div>
  </div>
</Fill>
`;

exports[`Holders render Square correctly 1`] = `
<Square
  align="center"
  fillerStyle={null}
  height={null}
  side={null}
  width={null}
>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <div
      style={
        Object {
          "background": undefined,
          "display": "inline-block",
          "height": null,
          "lineHeight": null,
          "textAlign": "center",
          "width": null,
        }
      }
    >
      foo-2
    </div>
  </div>
</Square>
`;

exports[`Holders render Square correctly 2`] = `
<Square
  align="center"
  color="red"
  fillerStyle={null}
  height={60}
  side={null}
  width={50}
>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <div
      style={
        Object {
          "background": "red",
          "display": "inline-block",
          "height": 50,
          "lineHeight": "50px",
          "textAlign": "center",
          "width": 50,
        }
      }
    >
      foo-2-1
    </div>
  </div>
</Square>
`;

exports[`Holders render Table correctly 1`] = `
<Table
  align="center"
  cols={2}
  fillerStyle={null}
  gap={2}
  height={null}
  rows={2}
  width={null}
>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <div
      style={
        Object {
          "background": "transparent",
          "display": "inline-block",
          "height": null,
          "position": "relative",
          "width": null,
        }
      }
    >
                
    </div>
  </div>
</Table>
`;

exports[`Holders render Table correctly 2`] = `
<Table
  align="center"
  color="red"
  cols={2}
  fillerStyle={null}
  gap={4}
  height={60}
  rows={2}
  width={360}
>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <div
      style={
        Object {
          "background": "transparent",
          "display": "inline-block",
          "height": 60,
          "position": "relative",
          "width": 360,
        }
      }
    >
                
      <div
        style={
          Object {
            "background": "red",
            "height": 28,
            "left": 0,
            "lineHeight": "28px",
            "overflow": "hidden",
            "position": "absolute",
            "textAlign": "center",
            "top": 0,
            "width": 178,
          }
        }
      >
        foo-3-1
      </div>
      <div
        style={
          Object {
            "background": "red",
            "height": 28,
            "left": 0,
            "lineHeight": "28px",
            "overflow": "hidden",
            "position": "absolute",
            "textAlign": "center",
            "top": 32,
            "width": 178,
          }
        }
      >
        foo-3-1
      </div>
      <div
        style={
          Object {
            "background": "red",
            "height": 28,
            "left": 182,
            "lineHeight": "28px",
            "overflow": "hidden",
            "position": "absolute",
            "textAlign": "center",
            "top": 0,
            "width": 178,
          }
        }
      >
        foo-3-1
      </div>
      <div
        style={
          Object {
            "background": "red",
            "height": 28,
            "left": 182,
            "lineHeight": "28px",
            "overflow": "hidden",
            "position": "absolute",
            "textAlign": "center",
            "top": 32,
            "width": 178,
          }
        }
      >
        foo-3-1
      </div>
    </div>
  </div>
</Table>
`;

exports[`Holders render Text correctly 1`] = `
<Text
  fillerStyle={null}
  fontSize="0.7em"
  height={null}
  length={100}
  lineHeight={2.3}
  width={null}
>
  <div
    style={
      Object {
        "background": "transparent",
        "fontSize": "0.7em",
        "height": null,
        "lineHeight": 2.3,
        "overflow": "hidden",
        "width": null,
      }
    }
  >
    <span
      style={
        Object {
          "background": undefined,
          "wordBreak": "break-word",
          "wordWrap": "break-word",
        }
      }
    >
                                                                                                                                                                                                              
    </span>
  </div>
</Text>
`;

exports[`Holders render Text correctly 2`] = `
<Text
  color="red"
  fillerStyle={null}
  fontSize="0.7em"
  height={null}
  length={300}
  lineHeight={2.3}
  width={null}
>
  <div
    style={
      Object {
        "background": "transparent",
        "fontSize": "0.7em",
        "height": null,
        "lineHeight": 2.3,
        "overflow": "hidden",
        "width": null,
      }
    }
  >
    <span
      style={
        Object {
          "background": "red",
          "wordBreak": "break-word",
          "wordWrap": "break-word",
        }
      }
    >
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              
    </span>
  </div>
</Text>
`;
